janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/util/SSLConfigurationCallback.java

```diff
@@ -0,0 +1,176 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.es.rest.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback;
+
+/**
+ * Use {@link Builder#create()} or {@link Builder#createCustom(SSLContextBuilder)} to create an instance of this callback
+ *
+ */
+public class SSLConfigurationCallback implements HttpClientConfigCallback {
+
+    private final String trustStoreFile;
+    private final String trustStorePassword;
+    private final String keyStoreFile;
+    private final String keyStorePassword;
+    private final String keyPassword;
+    private final SSLContextBuilder sslContextBuilder;
+    private final boolean disableHostNameVerification;
+    private final boolean allowSelfSignedCertificates;
+
+    private SSLConfigurationCallback(final SSLContextBuilder sslContextBuilder,
+            final String trustStoreFile,
+            final String trustStorePassword,
+            final String keyStoreFile,
+            final String keyStorePassword,
+            final String keyPassword,
+            final boolean disableHostNameVerification,
+            final boolean allowSelfSignedCertificates) {
+
+        this.sslContextBuilder = sslContextBuilder;
+
+        this.trustStoreFile = trustStoreFile;
+        this.trustStorePassword = trustStorePassword;
+
+        this.keyStoreFile = keyStoreFile;
+        this.keyStorePassword = keyStorePassword;
+        this.keyPassword = keyPassword;
+
+        this.disableHostNameVerification = disableHostNameVerification;
+        this.allowSelfSignedCertificates = allowSelfSignedCertificates;
+    }
+
+    @Override
+    public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
+        final SSLContext sslcontext;
+
+        final TrustStrategy trustStrategy = (allowSelfSignedCertificates ? new TrustSelfSignedStrategy() : null);
```

Comment:
why the parentheses around the ternary operation?

Reply:
Looks like without a good reason, thanks!